src/main/java/io/vertx/ext/stomp/impl/Subscription.java

```diff
@@ -0,0 +1,145 @@
+package io.vertx.ext.stomp.impl;
+
+import io.vertx.ext.stomp.Frame;
+import io.vertx.ext.stomp.StompServerConnection;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a subscription.
+ * This class is thread safe.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class Subscription {
```

Comment:
Also there's still the ability to DoS the server by never acking messages until out of memory occurs. Maybe add an issue for that? I don't want to prevent the initial PR getting into master.